src/ts/stringutils.ts

```diff
@@ -112,7 +112,37 @@ export namespace StringUtils {
         time = player.getCurrentTime();
       } else if (formatString.indexOf('adDuration') > -1) {
         time = player.getDuration();
+      } else if (formatString.indexOf('adBreakRemainingTime') > -1) { // To display the remaining time in the ad bread as opposed to in the ad
+        time = 0;
+        
+        // compute list of ads and index of active ad
+        if (player.ads.isLinearAdActive()) {
+          let scheduledAds = player.ads.getActiveAdBreak().ads;
+          let durations: number[] = [];
+          let activeAd = player.ads.getActiveAd();
+          let indexOfActiveAd: number = 0;
+  
+          for(let i=0; i<scheduledAds.length; i++) {
+            let iAd = scheduledAds[i];
+            if(iAd.isLinear) {
+              durations.push((iAd as LinearAd).duration);
+  
+              if(iAd.id == activeAd.id) {
+                indexOfActiveAd = i;
+              }
+            }
+          }
+          // Compute duration of ads to be played (incl active ad)
+          let duration = durations.slice(indexOfActiveAd).reduce((prev, current) => prev + current, 0);
```

Comment:
This could be simplified like this (unless I'm missing something):

```suggestion
          const duration = player.ads.getActiveAdBreak().ads
            .map(ad => ad.duration)
            .reduce((prev, current) => prev + current, 0);
```

Reply:
thanks @dweinber . I've simplified the code per your guidance.